crates/ruff/src/args.rs

```diff
@@ -491,97 +506,168 @@ impl From<&LogLevelArgs> for LogLevel {
     }
 }
 
+#[derive(Default)]
+pub struct ConfigArgs {
+    /// Path to a pyproject.toml or ruff.toml config file (etc.), if provided
+    config_file: Option<PathBuf>,
+    /// Overrides provided via the `--config FOO=BAR` option on the CLI
+    config_overrides: Configuration,
+    /// Overrides provided via specific (legacy-ish) flags such as --line-length etc
+    cli_overrides: CliOverrides,
+}
+
+impl ConfigArgs {
+    pub fn config_file(&self) -> Option<&PathBuf> {
+        self.config_file.as_ref()
+    }
+
+    fn from_cli_options(
+        config_options: Option<Vec<ConfigOption>>,
+        cli_overrides: CliOverrides,
+        isolated: bool,
+    ) -> Result<Self, anyhow::Error> {
+        let mut new = Self {
+            cli_overrides,
+            ..Self::default()
+        };
+
+        let Some(options) = config_options else {
+            return Ok(new);
+        };
+
+        for option in options {
+            match option {
+                ConfigOption::ConfigOverride(overridden_option) => {
+                    new.config_overrides = new.config_overrides.combine(
+                        Configuration::from_options(*overridden_option, &path_dedot::CWD)?,
+                    );
```

Comment:
We'll need to document the precedence of options. What binds higher: `--line-length` or `--config line-length`?

Reply:
Currently `--line-length`. Agree it needs to be documented!